src/dispatch/plugins/dispatch_google/docs/plugin.py

```diff
@@ -23,7 +24,80 @@ def remove_control_characters(s):
     return "".join(ch for ch in s if unicodedata.category(ch)[0] != "C")
 
 
-def replace_text(client: Any, document_id: str, replacements: List[str]):
+def find_links(obj: Any, find_key: str) -> Iterator[List[Any]]:
+    """Enumerate all the links found.
+    Returns a path of object, from leaf to parents to root.
+
+    This method was originally implemented in the open source library `Beancount`.
+    The original source code can be found at
+    https://github.com/beancount/beancount/blob/master/tools/transform_links_in_docs.py.
+    BeanCount is licensed under the GNU GPLv2.0 license.
+    """
+    if isinstance(obj, dict):
+        for key, value in obj.items():
+            if key == find_key:
+                yield [value, obj]
+            else:
+                for found in find_links(value, find_key):
+                    found.append(obj)
+                    yield found
+    elif isinstance(obj, list):
+        for value in obj:
+            for found in find_links(value, find_key):
+                found.append(obj)
+                yield found
+
+
+def iter_links(document: dict) -> List[Tuple[str, str]]:
```

Comment:
```suggestion
def iter_links(document: dict) -> list[tuple[str, str]]:
```
you don't need to import these from typing